packages/dashboard-frontend/src/store/GitOauthConfig/index.ts

```diff
@@ -28,88 +31,141 @@ import { AppThunk } from '..';
 export interface State {
   isLoading: boolean;
   gitOauth: IGitOauth[];
+  providersWithToken: api.GitOauthProvider[];
+  skipOauthProviders: api.GitOauthProvider[];
   error: string | undefined;
 }
 
 export enum Type {
-  REQUEST_GIT_OAUTH_CONFIG = 'REQUEST_GIT_OAUTH_CONFIG',
-  DELETE_OAUTH = 'DELETE_OAUTH',
-  RECEIVE_GIT_OAUTH_CONFIG = 'RECEIVE_GIT_OAUTH_CONFIG',
-  RECEIVE_GIT_OAUTH_CONFIG_ERROR = 'RECEIVE_GIT_OAUTH_CONFIG_ERROR',
+  REQUEST_GIT_OAUTH = 'REQUEST_GIT_OAUTH',
+  DELETE_GIT_OAUTH_TOKEN = 'DELETE_GIT_OAUTH_TOKEN',
+  RECEIVE_GIT_OAUTH_PROVIDERS = 'RECEIVE_GIT_OAUTH_PROVIDERS',
+  RECEIVE_SKIP_OAUTH_PROVIDERS = 'RECEIVE_SKIP_OAUTH_PROVIDERS',
+  DELETE_SKIP_OAUTH = 'DELETE_SKIP_OAUTH',
+  RECEIVE_GIT_OAUTH_ERROR = 'RECEIVE_GIT_OAUTH_ERROR',
 }
 
-export interface RequestGitOauthConfigAction extends Action {
-  type: Type.REQUEST_GIT_OAUTH_CONFIG;
+export interface RequestGitOAuthAction extends Action {
+  type: Type.REQUEST_GIT_OAUTH;
 }
 
 export interface DeleteOauthAction extends Action {
-  type: Type.DELETE_OAUTH;
+  type: Type.DELETE_GIT_OAUTH_TOKEN;
   provider: api.GitOauthProvider;
 }
 
-export interface ReceiveGitOauthConfigAction extends Action {
-  type: Type.RECEIVE_GIT_OAUTH_CONFIG;
-  gitOauth: IGitOauth[];
+export interface ReceiveGitOAuthConfigAction extends Action {
+  type: Type.RECEIVE_GIT_OAUTH_PROVIDERS;
+  supportedGitOauth: IGitOauth[];
+  providersWithToken: api.GitOauthProvider[];
 }
 
-export interface ReceivedGitOauthConfigErrorAction extends Action {
-  type: Type.RECEIVE_GIT_OAUTH_CONFIG_ERROR;
+export interface ReceivedGitOauthErrorAction extends Action {
+  type: Type.RECEIVE_GIT_OAUTH_ERROR;
   error: string;
 }
 
+export interface ReceiveSkipOauthProvidersAction extends Action {
+  type: Type.RECEIVE_SKIP_OAUTH_PROVIDERS;
+  skipOauthProviders: api.GitOauthProvider[];
+}
+
+export interface DeleteSkipOauth extends Action {
+  type: Type.DELETE_SKIP_OAUTH;
+  provider: api.GitOauthProvider;
+}
+
 export type KnownAction =
-  | RequestGitOauthConfigAction
+  | RequestGitOAuthAction
+  | ReceiveGitOAuthConfigAction
+  | ReceiveSkipOauthProvidersAction
   | DeleteOauthAction
-  | ReceiveGitOauthConfigAction
-  | ReceivedGitOauthConfigErrorAction;
+  | DeleteSkipOauth
+  | ReceivedGitOauthErrorAction;
 
 export type ActionCreators = {
+  requestSkipAuthorisationProviders: () => AppThunk<KnownAction, Promise<void>>;
   requestGitOauthConfig: () => AppThunk<KnownAction, Promise<void>>;
```

Comment:
As far as I can see, these actions are called in one place only (in the `GitServicesTab` component). Have you thought about merging these actions into one?